### PR TITLE
DEV: Use Ember.Service over manually registering a factory.

### PR DESCRIPTION
**Pull request by tgxworld**
Follow up to 9e827eb420c7aefd166fba1e7724eced81aaffe7

### CODE REVIEW

File: app/assets/javascripts/discourse/app/pre-initializers/inject-discourse-objects.js [hunk 1]
```diff
@@ -28,7 +28,10 @@ export default {
 
     const messageBus = window.MessageBus;
     app.register("message-bus:main", messageBus, { instantiate: false });
-    ALL_TARGETS.forEach(t => app.inject(t, "messageBus", "message-bus:main"));
+
+    ALL_TARGETS.concat("service").forEach(t =>
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/pre-initializers/inject-discourse-objects.js:32]: Is there a reason we can't add `service` to `ALL_TARGETS`?

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/pre-initializers/inject-discourse-objects.js:32]: We could but I figured I would just avoid injecting everything everywhere when we don't have to.

**Comment by tgxworld** [comment]
Thank you for reviewing @davidtaylorhq and bringing the use of a service up :)

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6 [hunk 2]
```diff
@@ -22,15 +17,7 @@ export default Component.extend({
...
-  presenceManager: null,
-
-  init() {
-    this._super(...arguments);
-
-    this.setProperties({
-      presenceManager: getOwner(this).lookup("presence-manager:main")
-    });
-  },
+  presenceManager: inject(),
```
Comment (eviltrout) [inline plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6:20]: Oh yeah that's much shorter :)

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6 [hunk 3]
```diff
@@ -1,17 +1,12 @@
 import Component from "@ember/component";
-import { getOwner } from "@ember/application";
 import { cancel } from "@ember/runloop";
 import { equal, gt } from "@ember/object/computed";
+import { inject } from "@ember/service";
```
Comment (CvX) [inline plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6:4]:
```suggestion
import { inject as service } from "@ember/service";
```

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6 (hunk 2 above)
Comment (CvX) [inline plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.js.es6:20]:
```suggestion
  presenceManager: service(),
```

File: plugins/discourse-presence/assets/javascripts/discourse/components/topic-presence-display.js.es6 [hunk 4]
```diff
@@ -1,17 +1,12 @@
 import Component from "@ember/component";
-import { getOwner } from "@ember/application";
 import { gt } from "@ember/object/computed";
+import { inject } from "@ember/service";
```
Comment (CvX) [inline plugins/discourse-presence/assets/javascripts/discourse/components/topic-presence-display.js.es6:3]:
```suggestion
import { inject as service } from "@ember/service";
```

File: plugins/discourse-presence/assets/javascripts/discourse/components/topic-presence-display.js.es6 [hunk 5]
```diff
@@ -1,17 +1,12 @@
...
 
 export default Component.extend({
   topic: null,
-  presenceManager: null,
-
-  init() {
-    this._super(...arguments);
-    this.set("presenceManager", getOwner(this).lookup("presence-manager:main"));
-  },
+  presenceManager: inject(),
```
Comment (CvX) [inline plugins/discourse-presence/assets/javascripts/discourse/components/topic-presence-display.js.es6:9]:
```suggestion
  presenceManager: service(),
```

**Pull request merged**